src/lib.rs

```diff
@@ -410,6 +410,35 @@ impl<'poll> Tiny<'poll> {
             }
         } else if words[0] == "clear" {
             self.tui.clear(&src.to_target());
+        } else if words[0] == "ignore" {
+            match src {
+                MsgSource::Serv { serv_name } => {
+                    self.tui.toggle_ignore(Some(Timestamp::now()),
+                        &MsgTarget::AllServTabs{
+                            serv_name: &serv_name,
+                        }
+                    );
+                }
+                MsgSource::Chan {
+                    serv_name,
+                    chan_name,
+                } => {
+                    self.tui.toggle_ignore(Some(Timestamp::now()),
+                        &MsgTarget::Chan{
+                            serv_name: &serv_name,
+                            chan_name: &chan_name,
+                        }
+                    );
+                }
+                MsgSource::User { serv_name, nick } => {
```

Comment:
This part is not quite right .. `MsgSource::User` means that the message has originated from a user tab. So this happens when you run `/ignore` in a private messaging tab. I think in this case we should just apply this to this users's tab, so `MsgTarget::User { serv_name: &serv_name, nick: &nick }` (just replace `AllUserTabs` with `User`).